freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/user/handler/SetPasswordHandler.java

```diff
@@ -31,14 +39,21 @@ public String selector() {
     @Override
     public void accept(Event<SetPasswordRequest> setPasswordRequestEvent) {
         SetPasswordRequest request = setPasswordRequestEvent.getData();
-        LOGGER.info("SetPasswordHandler accepting request {}", request);
+        Map<String, String> mdcContext = MDCBuilder.getMdcContextMap();
         try {
-            FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStackId(request.getResourceId());
+            Stack stack = stackService.getStackById(request.getResourceId());
+            MDCBuilder.buildMdcContext(stack);
+            LOGGER.info("SetPasswordHandler accepting request {}", request);
+
+            FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStack(stack);
             freeIpaClient.userSetPassword(request.getUsername(), request.getPassword());
             SetPasswordResult result = new SetPasswordResult(request);
             request.getResult().onNext(result);
         } catch (Exception e) {
             request.getResult().onError(e);
+        } finally {
+            MDCBuilder.cleanupMdc();
+            MDCBuilder.buildMdcContextFromMap(mdcContext);
```

Comment:
I don't think restoring the MdContext is necessary

Reply:
I see that the AsyncTaskExecutor is decorated to clean up the MDC. This one is invoked through the EventBus and I haven't tracked down whether the MDC is cleaned up automatically.
I'll investigate some more, but please let me know if you know the answer.

Reply:
done. I've also used the "flow" field to track the sync operation id.